test/modules/DeploymentTester/Settings.cs

```diff
@@ -25,33 +25,48 @@ public class Settings
                     configuration.GetValue<string>("IOTEDGE_DEVICEID"),
                     configuration.GetValue<string>("IOTEDGE_MODULEID"),
                     configuration.GetValue("DEPLOYMENT_TESTER_MODE", DeploymentTesterMode.Receiver),
-                    configuration.GetValue<string>("IOT_HUB_CONNECTION_STRING"),
+                    Option.Maybe(configuration.GetValue<string>("IOT_HUB_CONNECTION_STRING")),
                     configuration.GetValue("testResultCoordinatorUrl", new Uri("http://testresultcoordinator:5001")),
                     configuration.GetValue("testStartDelay", TimeSpan.FromMinutes(2)),
                     configuration.GetValue("testDuration", TimeSpan.FromHours(1)),
                     configuration.GetValue<string>("trackingId"),
-                    configuration.GetValue<string>("targetModuleId"),
+                    Option.Maybe(configuration.GetValue<string>("targetModuleId")),
                     configuration.GetValue("DEPLOYMENT_UPDATE_PERIOD", TimeSpan.FromMinutes(5)));
             });
 
         Settings(
             string deviceId,
             string moduleId,
             DeploymentTesterMode testMode,
-            string iotHubConnectionString,
+            Option<string> iotHubConnectionString,
             Uri testResultCoordinatorUrl,
             TimeSpan testStartDelay,
             TimeSpan testDuration,
             string trackingId,
-            string targetModuleId,
+            Option<string> targetModuleId,
             TimeSpan deploymentUpdatePeriod)
         {
             this.DeviceId = Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId));
             this.ModuleId = Preconditions.CheckNonWhiteSpace(moduleId, nameof(moduleId));
-            this.IoTHubConnectionString = Preconditions.CheckNonWhiteSpace(iotHubConnectionString, nameof(iotHubConnectionString));
             this.TestResultCoordinatorUrl = Preconditions.CheckNotNull(testResultCoordinatorUrl, nameof(testResultCoordinatorUrl));
             this.TrackingId = Preconditions.CheckNonWhiteSpace(trackingId, nameof(trackingId));
-            this.TargetModuleId = Preconditions.CheckNonWhiteSpace(targetModuleId, nameof(targetModuleId));
+
+            if (testMode == DeploymentTesterMode.Sender)
+            {
+                if (!iotHubConnectionString.HasValue)
```

Comment:
nit: You can use Preconditions.CheckArgument(!iotHubConnectionString.HasValue, nameof(iotHubConnectionString)). It will throw ArgumentException instead of ArgumentNullException